### ISSUE TITLE
Clarifying documentation on taking gradients through gradient operation

### ISSUE THREAD

**Issue by renesax14** [facebookresearch/higher#51]
I want the gradient operation to be part of my computation graph and backprop through it.

I was reading the [docs](https://higher.readthedocs.io/en/latest/toplevel.html) and it said:

> track_higher_grads: if True, during unrolled optimization the graph be
retained, and the fast weights will bear grad funcs, so as to permit
backpropagation through the optimization process. Setting this to
False allows the returned differentiable optimizer to be used in
"test mode", without potentially tracking higher order gradients.
This can be useful when running the training loop at test time,
e.g. in k-shot learning experiments, without incurring a significant
memory overhead.

which I found confusing. First without reading the actual code "fast weights" isn't something the user would know what that is (I still don't know and I'v read the code a non trivial amount of time).

Second `permit
backpropagation through the optimization process` is also ambiguous. This only makes me feel that it repairs the issue with weights for nn.modules not being leafs anymore (so it allows grads to flow). Great that makes things like:

```
w^<t> := w^<t-1> - eta * g^<t>
```

differentiable I presume. But it doesn't tell me if 

``
g^<t>.grad
```

has gradients itself. This should be knowable from just reading the docs.

So the main question:

**do we get backpropagation through gradient operations by setting `track_higher_grads=True` or not**?

----

It also mentions something about test time...it says:

> Setting
                this to False allows the differentiable optimizer to be used in
                "test mode", without potentially tracking higher order
                gradients

this makes me think that when its true we do take derivatives with respect to gradients...but I can't tell for sure from the current docs.

-----


Some relevant snipits of code I extract:

```
all_grads = _torch.autograd.grad(
loss,
grad_targets,
create_graph=self._track_higher_grads,
allow_unused=True # boo
)

for group, mapping in zip(self.param_groups, self._group_to_param_list):
for p, index in zip(group['params'], mapping):
if self._track_higher_grads:
new_params[index] = p
else:
new_params[index] = p.detach().requires_grad_()

```
